macros/edr/dbt_artifacts/upload_artifacts_to_table.sql

```diff
@@ -1,18 +1,39 @@
-{% macro upload_artifacts_to_table(table_relation, artifacts, flatten_artifact_callback, truncate_if_on_run_end, should_commit=False) %}
+{% macro upload_artifacts_to_table(table_relation, artifacts, flatten_artifact_callback, append=False, should_commit=False) %}
     {% set flatten_artifact_dicts = [] %}
     {% for artifact in artifacts %}
         {% set flatten_artifact_dict = flatten_artifact_callback(artifact) %}
-        {% if flatten_artifact_dict is not none %}
+        {% if flatten_artifact_dict is not none and elementary.dict_has_nonempty_values(flatten_artifact_dict) %}
             {% do flatten_artifact_dicts.append(flatten_artifact_dict) %}
         {% endif %}
     {% endfor %}
-    {% if truncate_if_on_run_end and elementary.is_on_run_end() %}
-      {% do dbt.truncate_relation(table_relation) %}
+
+    {% if append %}
+        {# In append mode, just insert, and no need to be atomic #}
+        {% do elementary.insert_rows(table_relation, flatten_artifact_dicts, should_commit, elementary.get_config_var('dbt_artifacts_chunk_size')) %}
+    {% else %}
+        {% do elementary.overwrite_artifacts_table(table_relation, flatten_artifact_dicts, should_commit) %}
     {% endif %}
+{% endmacro %}
+
+{% macro overwrite_artifacts_table(table_relation, flatten_artifact_dicts, should_commit) %}
+    {% do return(adapter.dispatch('overwrite_artifacts_table', 'elementary')(table_relation, flatten_artifact_dicts, should_commit)) %}
+{% endmacro %}
+
+{% macro default__overwrite_artifacts_table(table_relation, flatten_artifact_dicts, should_commit) %}
+    {# First upload everything to a temp table #}
+    {% set temp_table_suffix = modules.datetime.datetime.utcnow().strftime('__tmp_%y%m%d%H%M%S%f') %}
```

Comment:
Why not just `__tmp`?
The reason I'm asking is that Postgres has a maximum of 63 characters for table names so we'd generally rather avoid long table names.
It caused a bit of problems with the anomaly tests because certain models resolved to the same table names after truncation (`<TRUNCATED_TEST_ID>__metrics`) even though they weren't the same test.

```py
In [203]: len(datetime.datetime.utcnow().strftime('__tmp_%y%m%d%H%M%S%f'))
Out[203]: 24 (0x18)
```

Reply:
I think you're probably right, just did it as an extra measure of safety.
Also IIRC `dbt.make_temp_relation` in postgres already adds a date so there's no sense in adding it twice...

So I'll remove it.